online.codingblocks.com/components/AboutCourse/IntroductionCard.vue

```diff
@@ -29,115 +31,203 @@
-            <div class="info-item">
-            <div>Course Duration </div>
-            <div>6 Months</div>
-          </div> -->
-          </div>
-          
+
         </div>
       </div>
     </div>
   </div>
 </template>
 
 <script>
+import { mapState } from 'vuex'
+import { JsonApiDataStore, JsonApiDataStoreModel } from 'jsonapi-datastore'
+import config from '~/config.json'
 import RatingStars from './RatingStars.vue'
 export default {
+  modules: ['@nuxtjs/axios'],
   name: 'IntroductionCard',
   props: {
     course: {
       type: Object,
-      required: true
+      required: true,
+    },
+  },
+  data: function () {
+    return {
+      filledHeartClass : 'fas',
+      unfilledHeartClass : 'far',
+      filled: false,
     }
   },
   components: {
-    RatingStars
+    RatingStars,
   },
   computed: {
     visibleInstructors() {
       return this.course.instructors.slice(0, 2)
     },
-    visibleInstructorNames () {
-      return this.visibleInstructors.map(i => i.name).join(', ')
+    visibleInstructorNames() {
+      return this.visibleInstructors.map((i) => i.name).join(', ')
     },
-  }
+    ...mapState(['session']),
+  },
+  created: function () {
+    if(this.$store.state.user_course_wishlists.course_and_wishlist.has(this.course['id'])){
+      this.filled = true;
+    }else{
+      this.filled = false;
+    }
+
+  },
+  methods: {
+    async addToWishlists() {
+      const isAuthenticated = this.$store.state.session.isAuthenticated
+
+      var obj = new JsonApiDataStoreModel('user_course_wishlists')
+      obj.setAttribute('courseId', this.course['id'])
+      obj.setRelationship('courses', new JsonApiDataStoreModel('courses',this.course['id']));
+      obj = obj.serialize();
+
+      const ancor = document.getElementById('wishlist_i')
+      const btn = document.getElementById('wishlist_btn')
+
+      if (isAuthenticated) {
+        if (this.filled) {
+            let course_and_wishlist_map = new Map();
+            course_and_wishlist_map = this.$store.state.user_course_wishlists.course_and_wishlist;
+            const wishlist_id = course_and_wishlist_map.get(this.course['id']);
+            this.$axios.$delete(`user_course_wishlists/${wishlist_id}`)
+            .then((res)=>{
+              this.$store.commit('user_course_wishlists/delCourse',wishlist_id)     
+              this.filled = false
+            })
+            .catch((err)=>{
+               console.error(err);
+            })
+           
+        } else {
+          
+            await this.$axios.$post('user_course_wishlists', obj)
+            .then((res)=>{
+                this.$store.commit('user_course_wishlists/setMaps', [{courseId:res.data.relationships.course.data.id,id:res.data.attributes.id}])
+                this.filled = true
+            })
+            .catch((err)=>{
+                console.error(err)
+            })    
+        }
+      } else {
+        
+        const { url, clientId } = config[process.env.NODE_ENV].oneauth
```

Comment:
Use LoginRequiredButton Component. Since the same logic is already written there.